### ISSUE TITLE
Suggestion List

### ISSUE THREAD

**Issue by Zeffuro** [paissaheavyindustries/Triggernometry#10]
Hey there, making a list here of things that I'd like to see in Triggernometry.
I understand that there are going to be things that you'll simply not implement or will take quite a bit of time but it's all food for thought.

General options: 
- Allow us to copy more then one action at a time.
- Allow us to fire a trigger manually (add it in the context menu when rightclicking a trigger or something).

Relating to image auras:
- Add more display modes: 
1. Zoomed to fill respecting Aspect Ratio horizontally only.
2. Zoomed to fill respecting Aspect Ratio vertically only.

- Add a rotation property (0 - 360 degrees)
- Add support for animated PNG's
- Add a mask mode where you can input an images aura ID and overlay on top of that.
- Allow us to set the framerate/speed for GIF's

Relating to variables and unique identifiers: 
- Create a for loop action where we can execute other actions.
- Allow us to target certain triggers / folders or use a regex to deactivate certain auras (I want to avoid using deactivate all auras as much as possible).
- Allow us to set a vector (X and Y) variable that has the visual placement guide, this will probably greatly enhance making setting files and allowing newer users to move stuff around without manually setting 2 variables (A VERY COMMON QUESTION).
- Allow us to resize the columns in the View Variables window.
- Allow us to delete certain variables through the View Variables window (just letting us select one and allowing the delete key would be enough).
- Allow us to access the triggers display name or give us a way to set a trigger specific variable.

I think that's everything I can thing of for now.

**Comment by dscott-soltech** [comment]
I'll add a vote for the image rotation property -- I believe Locrian said it was a bit trickier to implement than we'd expect, but I can see a lot of uses for it.

**Comment by paissaheavyindustries** [comment]
Been thinking of the workload associated to these suggestions, here are some comments:

- Allow us to copy more then one action at a time.
Can be done

- Allow us to fire a trigger manually (add it in the context menu when rightclicking a trigger or something).
Can be done

- Zoomed to fill respecting Aspect Ratio horizontally only.
Can be done

- Zoomed to fill respecting Aspect Ratio vertically only.
Can be done

- Add a rotation property (0 - 360 degrees)
Very complicated to add due to GDI, will try to add but no timeframe

- Add support for animated PNG's
Will be more complicated again due to GDI and WinForms, will try to add but no timeframe

- Add a mask mode where you can input an images aura ID and overlay on top of that.
Likely not going to add exactly something like this, due to the future layout mode plan already having something to this effect

- Allow us to set the framerate/speed for GIF's
Unlikely to add in the foreseeable future due to GDI GIF playback being handled by WinForms, and there is no way to set playback speed

- Create a for loop action where we can execute other actions.
Some form of loop action would be nice, will consider adding

- Allow us to target certain triggers / folders or use a regex to deactivate certain auras
Regex will be added for sure

- Allow us to set a vector (X and Y) variable...
Likely not going to add because 1) this is not trivial to add, and 2) the future layout mode will render all these funky additions irrelevant (it will be much easier for people to move things around as well)

- Allow us to resize the columns in the View Variables window.
Should be doable

- Allow us to delete certain variables through the View Variables window (just letting us select one and allowing the delete key would be enough).
Should be doable

- Allow us to access the triggers display name or give us a way to set a trigger specific variable.
Will add special variables to get trigger name and trigger GUID (and then you are able to use the GUID as part of your variable names, to create variables specific to that trigger)

**Comment by Aho-Senpai** [comment]
what about this ? (from Lombrax)
"Is there any chance we could get a z-coordinate for auras?
That way when I alt tab from the game, the auras don't redraw in the wrong order? (image over text instead of text over image)"

**Comment by paissaheavyindustries** [comment]
Yeah I am planning to add the z-coordinate too, in the most recent commit I did a temporary fix that will prevent them from going out of order when tabbing in and out, the order is preserved as is.

**Comment by Zeffuro** [comment]
Looking forward to all of those changes!
